pkg/prometheus/operator.go

```diff
@@ -1775,8 +1775,11 @@ func (c *Operator) createOrUpdateConfigurationSecret(ctx context.Context, p *mon
 	}
 	if p.Spec.Alerting != nil {
 		for i, am := range p.Spec.Alerting.Alertmanagers {
+			if err := store.AddBasicAuth(ctx, p.GetNamespace(), am.BasicAuth, fmt.Sprintf("alertmanager/auth/%d", i)); err != nil {
+				return errors.Wrapf(err, "alerting")
+			}
 			if err := store.AddSafeAuthorizationCredentials(ctx, p.GetNamespace(), am.Authorization, fmt.Sprintf("alertmanager/auth/%d", i)); err != nil {
-				return errors.Wrapf(err, "apiserver config")
+				return errors.Wrapf(err, "alerting")
```

Comment:
Why changing here?

Reply:
Previously it was the copy&paste of the error message when we set authorization for `p.Spec.APIServerConfig` ([here](https://github.com/prometheus-operator/prometheus-operator/blob/311d636b7c9314856dfedd091bcc9985b37835b9/pkg/prometheus/operator.go#L1768-L1775)) since here we are setting authorization for `p.Spec.Alerting` I thought it made more sense like this.